tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/quickfix/LiferayDependencyQuickFix.java

```diff
@@ -0,0 +1,269 @@
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ui.text.java.IInvocationContext;
+import org.eclipse.jdt.ui.text.java.IJavaCompletionProposal;
+import org.eclipse.jdt.ui.text.java.IProblemLocation;
+import org.eclipse.jdt.ui.text.java.IQuickFixProcessor;
+import org.eclipse.jdt.ui.text.java.correction.CUCorrectionProposal;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Display;
+import org.osgi.framework.Version;
+
+/**
+ * @author Simon Jiang
+ */
+
+@SuppressWarnings( "restriction" )
+public class LiferayDependencyQuickFix implements IQuickFixProcessor
+{
+
+    @Override
+    public boolean hasCorrections( ICompilationUnit unit, int problemId )
+    {
+        switch( problemId )
+        {
+        case IProblem.ImportNotFound:
+        case IProblem.UndefinedType:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    @Override
+    public IJavaCompletionProposal[] getCorrections( IInvocationContext context, IProblemLocation[] locations )
+        throws CoreException
+    {
+        if( locations == null || locations.length == 0 )
+        {
+            return null;
+        }
+
+        List<IJavaCompletionProposal> resultingCollections = new ArrayList<>();
+
+        for( int i = 0; i < locations.length; i++ )
+        {
+            IProblemLocation curr = locations[i];
+
+            process( context, curr, resultingCollections );
+        }
+
+        return resultingCollections.toArray( new IJavaCompletionProposal[resultingCollections.size()] );
+    }
+
+    private void process(
+        IInvocationContext context, IProblemLocation problem, List<IJavaCompletionProposal> proposals )
+    {
+        int id = problem.getProblemId();
+
+        if( id == 0 )
+        {
+            return;
+        }
+
+        switch( id )
+        {
+            case IProblem.ImportNotFound:
+                importNotFoundProposal( context, problem, proposals );
+                break;
+            case IProblem.UndefinedType:
+                undefinedType( context, problem, proposals );
+                break;
+            default:;
+        }
+    }
+
+    private void importNotFoundProposal(
+        IInvocationContext context, IProblemLocation problem, Collection<IJavaCompletionProposal> proposals )
+    {
+        ASTNode selectedNode = problem.getCoveringNode( context.getASTRoot() );
+
+        if( selectedNode == null )
+        {
+            return;
+        }
+
+        ImportDeclaration importDeclaration =
+            (ImportDeclaration) ASTNodes.getParent( selectedNode, ASTNode.IMPORT_DECLARATION );
+
+        if( importDeclaration == null )
+        {
+            return;
+        }
+
+        String importName = importDeclaration.getName().toString();
+        List<String> serviceWrapperList;
+        List<String> servicesList;
+        boolean depWrapperCanFixed = false;
```

Comment:
I will check it. This part code just copy from QuickFixGradleDep existed code and I used most of this class code and make it suport gradle and maven.